### PR TITLE
Don't claim to provide the extra ''

### PR DESCRIPTION
**Pull request by wimglenn**
Resolves https://github.com/pypa/setuptools/issues/1292

### CODE REVIEW

**Comment by di** [comment]
@wimglenn Sorry #1286 broke things for you. I think you probably want to move the check farther up and just never add the "empty" extra to `self.metadata.provides_extras`:

```diff
                 # Since this gets called multiple times at points where the
                 # keys have become 'converted' extras, ensure that we are only
                 # truly adding extras we haven't seen before here.
-                self.metadata.provides_extras.add(extra.split(':')[0])
+                extra = extra.split(':')[0]
+                if extra:
+                    self.metadata.provides_extras.add(extra)

         self._convert_extras_requirements()
         self._move_install_requirements_markers()
```

**Comment by wimglenn** [comment]
@di  Thanks, I suspected it could have been handled earlier but had trouble to find the source and wasn't sure if this was distutils fault somehow. I'm not too familiar with setuptools and this code is a little hard to follow with the module level "self" function and the monkeypatches. I've just edited the PR in the online github UI, hope that works OK.
